tremor-pipeline/src/op/qos/wal.rs

```diff
@@ -300,17 +335,27 @@ impl Operator for Wal {
                         // This is not for us
                         return;
                     };
-                self.confirmed.set(event_id);
-                if let Err(e) = self.state_tree.insert("read", self.confirmed) {
+
+                trace!("WAL confirm: {}", event_id);
+                let confirmed = if let Some(confirmed) = self.confirmed.as_mut() {
```

Comment:
could we avoid this with 'get_or_insert'?

Reply:
Sooooo much better!

Reply:
aaah, no, doesnt work, the mut reference makes the borrow checker unhappy.

Reply:
waht's the issue the borrow checker is throwing? something that might be solvable?

Reply:
we get a `&mut` reference back and this is consumed in `self.state_tree.insert(...)` and thus cannot be used in `events_tree.get()` later on.

Reply:
would `let confirmed: & <whatevertypeitis> = confirmed` work to downgrade it from a mutable to a immutable reference?

Reply:
it does! :heart: Man, this shadowing is really dirty.